//Question -3  Exception Handling in Java

import java.io.*;
import java.util.*;

public class Solution {

    public static void main(String[] args) {
    Scanner sc=new Scanner(System.in);
    try{
        int x=sc.nextInt();
        int y=sc.nextInt();
        System.out.println(x/y);
    }
    catch(ArithmeticException e1)
    {
        System.out.println("java.lang.ArithmeticException: / by zero");
    }
    catch(InputMismatchException e2)
    {
        System.out.println("java.util.InputMismatchException");
    }
    sc.close();
    }
}

-----------------------------------------------------------------------------

// Question 4 Exception Handling in Java

class MyCalculator {
   public long power(int n, int p) throws Exception{
    long result;
    if(n==0 && p==0){
        throw new Exception("n and p should not be zero.");
    }
    if(n<0 || p<0){
        throw new Exception("n or p should not be negative.");
    }
         result=(long)Math.pow(n,p);
         return result;
   }
    
}